pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -303,9 +304,84 @@ func resolveDataWithTemplateVars(ctx context.Context, data integration.Data, svc
 	return yaml.Marshal(&tree)
 }
 
-var varPattern = regexp.MustCompile(`‰(.+?)(?:_(.+?))?‰`)
+var ipv6Re = regexp.MustCompile(`^[0-9a-f:]+$`)
 
 func resolveStringWithTemplateVars(ctx context.Context, in string, svc listeners.Service) (out interface{}, err error) {
+	isThereAnIPv6Host := false
+
+	adHocTemplateVars := make(map[string]variableGetter)
+	for k, v := range templateVariables {
+		if k == "host" {
+			adHocTemplateVars[k] = func(ctx context.Context, tplVar string, svc listeners.Service) (string, error) {
+				host, err := getHost(ctx, tplVar, svc)
+				if ipv6Re.MatchString(host) {
+					isThereAnIPv6Host = true
+					if tplVar != "" {
+						return fmt.Sprintf("‰host_%s‰", tplVar), nil
```

Comment:
This format string has two % verbs but only one variable

Reply:
I think the first % should not be a verb. but instead `\%`